0017-negotiation-and-execution-protocol.adoc

```diff
@@ -0,0 +1,544 @@
+
+* Maker: Creates offers
+* Taker: Takes offers
+
+Note that these roles are orthogonal to the cryptographic roles (Alice and Bob) in the execution protocol.
+Both the maker and the taker can theoretically be in the role of Alice (who generates the secret) or Bob.
+During the negotiation phase both parties have to reach agreement on who will be in the role of Alice and who will be in the role of Bob.
+
+==== Bid and Ask ====
+
+In the context of this document Bid and Ask is always from the market maker point of view.
+
+* Bid: The amount of asset the maker offers.
+* Ask: The amount of asset the maker asks for from a potential taker.
+
+
+=== Recommendation for protocol changes
+
+The general idea is to first define what parameters are needed for the execution phase and then design a negotiation protocol that outputs these parameters for the execution.
+
+As RFC003 is currently the only implementation of an execution protocol we focus on this protocol.
+Once more execution protocols are defined we will have to define more negotiation protocols according to the parameters required by the execution protocol.
+This means, that some negotiation protocols could be re-used for multiple execution protocols if the execution protocols have the same input parameters.
+
+=== Invitation phase
+
+We introduce an invitation phase in order to enable decentralization.
+Makers don't necessarily have a central order book, the invitation helps them to overcome the problem of decentral offers getting stale.
+
+On an abstract level an invitation protocol can be defined as a function:
+
+```
+fn invitation() -> negotiation_params[] {
+    // steps (messages) needed to come from nothing to negotiation_params
+}
+```
+
+==== Communication
+
+Message Overview:
+
+![Invitation](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/comit-network/spikes/master/assets/0017-SequenceDiagram-InvitationPhase.puml&fmt=svg)
+
+==== Trade Invitation ====
+
+The trade invitation specifies which assets on which ledgers a maker _is willing_ to trade.
+The trade invitation is not necessarily part of the negotiation.
+Its purpose is to connect maker and taker.
+
+An asset-availability is defined by:
+
+[%header,cols=1*]
+|===
+|required
+
+|connection-info
+|bid-asset
+|ask-asset
+|===
+
+[%header,cols=1*]
+|===
+|optional
+
+|timestamp
+|bid-amount
+|ask-amount
+|===
+
+Constraints:
+
+* Connection-info currently consists of IP-address and libp2p peer-id
+* The optional parameters help takers to understand what a maker was capable of at a certain point in time.
+
+Representation in JSON:
+
+```json
+{
+    "connection-info": { ... },
+    "bid": {
+        "ledger": ...
+        "asset": ...
+    },
+    "ask": {
+        "ledger": ...
+        "asset": ...
+    }
+}
+```
+When designing the negotiation protocol
+One should aim for a standardized format for trade invitations to be able to process them in the negotiation phase.
```

Comment:
I was going to say just trading pair of assets.  Assets have a symbol (BTC), a name/description (bitcoin), and with @bonomat's comment a ledger (perhaps this can be empty for symbols that exist on a single ledger only.